### PR TITLE
SBOM Report integration in File Page

### PR DESCRIPTION
**Pull request by Yibaebi**

### CODE REVIEW

File: app/components/file/report-drawer/sbom-reports/sample/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+export default class FileReportDrawerSbomReportsSampleComponent extends Component {
+  @service declare network: NetworkService;
+  @service declare intl: IntlService;
+  @service('notifications') declare notify: NotificationService;
+  @service('browser/window') declare window: Window;
+
+  @tracked hasContactedSupport = false;
+
+  SUPPORT_ENDPOINT = 'v2/sample_sbom/contact_support';
+  REPORT_ENDPOINT = 'v2/sample_sbom/sample-reports';
```
Comment (future-pirate-king) [inline app/components/file/report-drawer/sbom-reports/sample/index.ts:18]: this should be property in environment variable

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/adapters/file.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  urlForQuery(query: ProjectFilesQuery, modelName: string | number) {
+    return this._buildNestedURL(modelName, query.projectId);
+  }
+
+  async getRelatedFileSbFile(fileId: string | number) {
+    const fileURL = this._buildURL('file', fileId);
+    const URL = `${fileURL}/sb_file`;
+
+    const sbFileRes = await this.network.request(URL);
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/adapters/file.ts:37]: @cosmosgenius Mentioned that we should slowly migrate to the network service.

[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/adapters/file.ts:37]: There is no problem with the Ajax service.

[the remark this answers is not in the document]
Comment (cosmosgenius) [reply app/adapters/file.ts:37]: wait here i think the best is `ajax` and not use the network... ajax is better and migration to future ember systems is easier...
Anything model, adapter or store related its better to use `this.ajax` which is not a injected service...
If you are doing a network call for something else like from component directly network is better.

The reason being the custom ajax service is old and not updated... but in store `this.ajax` internal uses fetch and is modern...

**Review by cosmosgenius — CHANGES_REQUESTED** [review]

File: app/components/file/report-drawer/sbom-reports/sample/index.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import SbomReportModel, { SbomReportType } from 'irene/models/sbom-report';
+import parseError from 'irene/utils/parse-error';
+
+type SbomScanReportQueryResponse =
+  DS.AdapterPopulatedRecordArray<SbomReportModel> & {
+    meta: { count: number };
+  };
+
+export default class FileReportDrawerSbomReportsSampleComponent extends Component {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (future-pirate-king) [inline app/components/file/report-drawer/sbom-reports/sample/index.ts:20]: let it show warning will help us fix later

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by cosmosgenius — APPROVED** [review]

**Pull request merged**